db/interactive_client.go

```diff
@@ -69,11 +77,28 @@ func (c *InteractiveClient) InteractiveQuery(resultsStreamer *results.ResultStre
 
 		// wait for the resultstreamer to have streamed everything out
 		// so that we know
-		resultsStreamer.Wait()
+		c.resultsStreamer.Wait()
 	}
+
+	// close up the SIGINT channel so that the receiver goroutine can quit
+	close(interruptSignalChannel)
+}
+
+func (c *InteractiveClient) startInterruptListener() chan os.Signal {
```

Comment:
should this also be startCancelHandler? or the other one should be startInterruptListener